cohort-engine/src/main/java/com/ibm/cohort/engine/measure/cache/RetrieveCacheContext.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.cohort.engine.measure.cache;
+
+import com.github.benmanes.caffeine.cache.Cache;
+
+public interface RetrieveCacheContext {
+
+	Cache<CacheKey, Iterable<Object>> newCache(String patientId);
+
+	Cache<CacheKey, Iterable<Object>> getCurrentCache();
+
+	// TODO: Is this where the cache would be persisted to a datastore???
```

Comment:
Would you want a `RetrieveCacheContext` to know how to persist itself, or would you want a separate class that would know how to take a `RetrieveCacheContext` and persist it somewhere?

Reply:
My current thought process was to have the context handle both the loading from and persisting to the datastore.  We could, for example, have a `COSRetrieveCacheContext` or a `DB2RetrieveCacheContext`.  I'm not sure if that's too much responsibility for one class though.